review-problems.ml

```diff
@@ -676,7 +790,21 @@ module Functions = struct
 
      Rank: **
    *)
-  let poly_gen cs = assert false
+  let poly_gen cs = match cs with
+    | y::ys ->
+      (fun x ->
+         match
+           begin
+             List.fold_left
```

Comment:
This is not really what the question is asking for. The question is asking for `poly_gen cs` to compute a non-recursive function `f` that calculates the polynomial evaluated as `x`. In order words, the returned function should not be doing recursion *at all*. Your solution is doing recursion on `ys` via `fold_left`.

Reply:
Sorry about that, I fixed it now and pushed the new commit.